packages/kit/types/index.d.ts

```diff
@@ -115,8 +115,14 @@ export interface HandleError {
  *
  * Note that you can use [generated types](/docs/types#generated-types) instead of manually specifying the Params generic argument.
  */
-export interface Load<Params = Record<string, string>, Props = Record<string, any>> {
-	(input: LoadInput<Params>): MaybePromise<Either<Fallthrough, LoadOutput<Props>>>;
+export interface Load<
+	Params = Record<string, string>,
+	InputProps = Record<string, any>,
+	OutputProps = Record<string, any>
+> {
+	(input: LoadInput<Params, InputProps>): MaybePromise<
+		Either<Fallthrough, LoadOutput<OutputProps>>
+	>;
```

Comment:
It would be good if we could omit `OutputProps` in the common case where they're just being passed through, I think?

```suggestion
export interface Load<
	Params extends Record<string, string> = Record<string, string>,
	InputProps extends Record<string, any> = Record<string, any>,
	OutputProps extends Record<string, any> = InputProps
> {
	(input: LoadInput<Params, InputProps>): MaybePromise<
		Either<Fallthrough, LoadOutput<OutputProps>>
	>;
```

Reply:
In fact, I think this would make it a non-breaking change?

Reply:
Would still be a breaking change.
Before this PR `props` (`InputProps`) could hold any `Record`, If someone would have set the `Props` before (`OutputProps`), this would now become the `InputProps` so this could throw some errors.
Here is an example

Before:
```ts
<script lang="ts" context="module">
   import type { Load } from '@sveltejs/kit'

   type Props = { value: string } 

   export const load: Load<any, Props> = async ({ props }) => {

      console.log(props.something) // valid

      return { /* ... */ }
   }
</script>
```
After:
```ts
<script lang="ts" context="module">
   import type { Load } from '@sveltejs/kit'

   type Props = { value: string } 

   export const load: Load<any, Props> = async ({ props }) => {

      console.log(props.something) // not valid anymore because only `value` is a known property

      return { /* ... */ }
   }
</script>
```

Reply:
Ah yep, gotcha. I've updated the changeset to reflect that. Breakingness aside, does this suggestion make sense?

Reply:
Yes, they make sense. I haven't spotted the missing `extends` in the generics because I was lazy and just copied the `Props` code from somewhere else ^^
Maybe it would make sense to take another look and update places where the `extends` is missing in the current code base? Should I take a look and include the fixes in this PR?

Also having the same `InputProps` and `OutputProps` could be a valid case 👍

Reply:
There's probably some other places that should use `extends`, but I'll merge this PR in the meantime anyway. Thank you!